CONTRIBUTING.md

```diff
@@ -0,0 +1,257 @@
+
+Avoid abbreviations in class and variable names. A good rule of thumb is that you should only use
+them if you would also do so verbally, for example "id" and "config" are probably reasonable.
+Single-letter variables are permissible if the variable scope is only a few lines, or for commonly
+understood cases (like `i` for a loop index).
+
+Keep source files short. Short files are easy to understand and test. The average should probably 
+be around 200-300 lines. 
+
+### Javadoc
+
+All types in "API" packages must be documented. For "internal" packages, documentation is optional,
+but in no way discouraged: it's generally a good idea to have a class-level comment that explains
+where the component fits in the architecture, and anything else that you feel is important.
+
+### Logs
+
+We use SLF4J; loggers are declared like this:
+
+```java
+private static final Logger LOG = LoggerFactory.getLogger(TheEnclosingClass.class);
+```
+
+
+### Nullability annotations
+
+We use the [Spotbugs annotations](https://spotbugs.github.io) to document nullability of parameters,
+method return types and class members.
+
+Please annotate any new class or interface with the appropriate annotations: `@NonNull`, `@Nullable`. Make sure you import 
+the types from `edu.umd.cs.findbugs.annotations`, there are homonyms in the classpath.
+
+
+## Coding style -- test code
+
+Static imports are permitted in a couple of places:
+* All AssertJ methods, e.g.:
+  ```java
+  assertThat(node.getDatacenter()).isNotNull();
+  fail("Expecting IllegalStateException to be thrown");
+  ```
+* All Mockito methods, e.g.:
+  ```java
+  when(codecRegistry.codecFor(DataTypes.INT)).thenReturn(codec);
+  verify(codec).decodePrimitive(any(ByteBuffer.class), eq(ProtocolVersion.DEFAULT));
+  ```
+
+Test methods names use lower snake case, generally start with `should`, and clearly indicate the
+purpose of the test, for example: `should_fail_if_key_already_exists`. If you have trouble coming 
+up with a simple name, it might be a sign that your test does too much, and should be split.
+
+We use AssertJ (`assertThat`) for assertions. Don't use JUnit assertions (`assertEquals`, 
+`assertNull`, etc).
+
+Don't try to generify at all cost: a bit of duplication is acceptable, if that helps keep the tests
+simple to understand (a newcomer should be able to understand how to fix a failing test without
+having to read too much code).
+
+Test classes can be a bit longer, since they often enumerate similar test cases. You can also
+factor some common code in a parent abstract class named with "XxxTestBase", and then split
+different families of tests into separate child classes.
+
+## Running the tests
+
+### Unit tests
+
+    mvn clean test
+   
+### Integration tests
+
+    mvn clean verify
+
+### Integration tests native
+
+    mvn clean verify -Dnative
+    
+### Generating documentation
+   
+    mvn clean package -Prelease    
+    
+
+## Build
```

Comment:
Overall this section doesn't introduce anything new. I also find it a bit strange that we instruct people to run `mvn clean install -DskipTests`. I suggest that we remove this section completely.

Reply:
this is also a copy-paste from the quarks web page, I'll remove it